### PR TITLE
Goliaths drop plates at the ashwalker tendril

### LINKED ISSUE #47593: Goliaths won’t drop their hide if they were sacrificed to the ashwalker nest

### PR DESCRIPTION
**Pull request by kriskog**
## About The Pull Request
simple_animal/gib() dropped butcher_results but not guaranteed_butcher_results

## Why It's Good For The Game
Fixes #47593

## Changelog
:cl: Skoglol
fix: Goliaths sacrificed at the ashwalker tendril now drop a hide plate.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 1]
```diff
@@ -310,11 +310,17 @@
 		adjustHealth(unsuitable_atmos_damage)
 
 /mob/living/simple_animal/gib()
+	var/atom/Tsec
 	if(butcher_results)
-		var/atom/Tsec = drop_location()
+		Tsec = drop_location()
```
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: You do this for both when you could do it just once

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 2]
```diff
@@ -310,11 +310,17 @@
...
 		for(var/path in butcher_results)
 			for(var/i in 1 to butcher_results[path])
 				new path(Tsec)
+	if(guaranteed_butcher_results)
+		Tsec = drop_location()
+		for(var/path in guaranteed_butcher_results)
```
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: for some cleaner code you could do
`for(var/path in guaranteed_butcher_results + butcher_results)`
And combine these two blocks

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/simple_animal.dm (hunk 2 above)
Comment (kriskog) [reply code/modules/mob/living/simple_animal/simple_animal.dm]: Yeah thats nicer, thanks

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 3]
```diff
@@ -310,11 +310,10 @@
 		adjustHealth(unsuitable_atmos_damage)
 
 /mob/living/simple_animal/gib()
-	if(butcher_results)
+	if(butcher_results || guaranteed_butcher_results)
 		var/atom/Tsec = drop_location()
-		for(var/path in butcher_results)
-			for(var/i in 1 to butcher_results[path])
```
Comment (ninjanomnom) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: You dropped this

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/simple_animal.dm (hunk 3 above)
Comment (kriskog) [reply code/modules/mob/living/simple_animal/simple_animal.dm]: REEEE coders nerfing mining again!

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**